### PR TITLE
Update to 2021.01.28

### PR DESCRIPTION
**Pull request by ellio167**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ellio167** [comment]
@conda-forge-admin, please rerender

**Comment by ellio167** [comment]
@jan-janssen I guess I could use some help here...

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
```
CMake Warning at /home/conda/feedstock_root/build_artifacts/openkim-models_1612821876891/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/share/cmake/kim-api/kim-api-pre-2.2-compatibility.cmake:51 (message):
  

  Pre 2.2.0 KIM-API package behavior is deprecated and will be removed in
  3.0.0.  Please update your CMakeLists.txt file to use a find_package()
  signature similar to one of the following:

     find_package(KIM-API 2.2 REQUIRED CONFIG)
     find_package(KIM-API-ITEMS 2.2 REQUIRED CONFIG)

  See the examples in the KIM API source package for more details.

Call Stack (most recent call first):
  /home/conda/feedstock_root/build_artifacts/openkim-models_1612821876891/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/share/cmake/kim-api/kim-api-config.cmake:52 (include)
  model-drivers/DUNN__MD_292677547454_000/CMakeLists.txt:37 (find_package)
```

**Comment by jan-janssen** [comment]
With the patch I get the following error: 
```
CMake Error at model-drivers/DUNN__MD_292677547454_000/CMakeLists.txt:52 (add_kim_api_model_driver_library):
  Unknown CMake command "add_kim_api_model_driver_library".
```

**Comment by jan-janssen** [comment]
For the last working builds when using cmake `3.18.2` it says: 
```
2020-10-09T01:59:41.8255944Z -- Found KIM-API: kim-api (Required is at least version "2.0") 
2020-10-09T01:59:41.9297047Z -- The CXX compiler identification is GNU 9.3.0
2020-10-09T01:59:41.9962075Z -- The C compiler identification is GNU 9.3.0
2020-10-09T01:59:42.1147308Z -- The Fortran compiler identification is GNU 9.3.0
2020-10-09T01:59:42.1416198Z -- Detecting CXX compiler ABI info
2020-10-09T01:59:42.2352681Z -- Detecting CXX compiler ABI info - done
2020-10-09T01:59:42.2636824Z -- Check for working CXX compiler: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++ - skipped
2020-10-09T01:59:42.2638573Z -- Detecting CXX compile features
2020-10-09T01:59:42.2648683Z -- Detecting CXX compile features - done
2020-10-09T01:59:42.2705120Z -- Detecting C compiler ABI info
2020-10-09T01:59:42.3630884Z -- Detecting C compiler ABI info - done
2020-10-09T01:59:42.3924002Z -- Check for working C compiler: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc - skipped
2020-10-09T01:59:42.3931616Z -- Detecting C compile features
2020-10-09T01:59:42.3940404Z -- Detecting C compile features - done
2020-10-09T01:59:42.4001792Z -- Detecting Fortran compiler ABI info
2020-10-09T01:59:44.5595032Z -- Detecting Fortran compiler ABI info - done
2020-10-09T01:59:44.5850594Z -- Check for working Fortran compiler: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gfortran - skipped
2020-10-09T01:59:44.5858653Z -- Checking whether $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gfortran supports Fortran 90
2020-10-09T01:59:44.7287021Z -- Checking whether $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gfortran supports Fortran 90 - yes
```

**Comment by jan-janssen** [comment]
That is the same output I get now after applying the patch:
```
-- The CXX compiler identification is GNU 9.3.0
-- The C compiler identification is GNU 9.3.0
-- The Fortran compiler identification is GNU 9.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gfortran - skipped
-- Checking whether $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gfortran supports Fortran 90
-- Checking whether $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gfortran supports Fortran 90 - yes
```

**Comment by jan-janssen** [comment]
In contrast without the patch:
```
2021-02-08T22:07:46.4210736Z CMake Warning at /home/conda/feedstock_root/build_artifacts/openkim-models_1612821876891/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/share/cmake/kim-api/kim-api-pre-2.2-compatibility.cmake:51 (message):
2021-02-08T22:07:46.4212259Z   
2021-02-08T22:07:46.4212731Z 
2021-02-08T22:07:46.4213943Z   Pre 2.2.0 KIM-API package behavior is deprecated and will be removed in
2021-02-08T22:07:46.4214620Z   3.0.0.  Please update your CMakeLists.txt file to use a find_package()
2021-02-08T22:07:46.4215166Z   signature similar to one of the following:
2021-02-08T22:07:46.4215464Z 
2021-02-08T22:07:46.4216062Z      find_package(KIM-API 2.2 REQUIRED CONFIG)
2021-02-08T22:07:46.4216776Z      find_package(KIM-API-ITEMS 2.2 REQUIRED CONFIG)
2021-02-08T22:07:46.4217126Z 
2021-02-08T22:07:46.4217721Z   See the examples in the KIM API source package for more details.
2021-02-08T22:07:46.4218003Z 
2021-02-08T22:07:46.4218367Z Call Stack (most recent call first):
2021-02-08T22:07:46.4219785Z   /home/conda/feedstock_root/build_artifacts/openkim-models_1612821876891/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/share/cmake/kim-api/kim-api-config.cmake:52 (include)
2021-02-08T22:07:46.4221036Z   model-drivers/DUNN__MD_292677547454_000/CMakeLists.txt:37 (find_package)
2021-02-08T22:07:46.4221363Z 
2021-02-08T22:07:46.4221556Z 
2021-02-08T22:07:46.5102878Z -- The CXX compiler identification is unknown
2021-02-08T22:07:46.6278972Z -- The C compiler identification is unknown
2021-02-08T22:07:46.6594193Z -- The Fortran compiler identification is unknown
2021-02-08T22:07:46.6630852Z CMake Error at model-drivers/DUNN__MD_292677547454_000/CMakeLists.txt:40 (project):
2021-02-08T22:07:46.6631672Z   The CMAKE_CXX_COMPILER:
2021-02-08T22:07:46.6631937Z 
2021-02-08T22:07:46.6632738Z     /home/conda/feedstock_root/build_artifacts/kim-api_1607958627046/_build_env/bin/x86_64-conda-linux-gnu-c++
2021-02-08T22:07:46.6633126Z 
2021-02-08T22:07:46.6633521Z   is not a full path to an existing compiler tool.
2021-02-08T22:07:46.6633781Z 
2021-02-08T22:07:46.6634189Z   Tell CMake where to find the compiler by setting either the environment
2021-02-08T22:07:46.6634743Z   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
2021-02-08T22:07:46.6635292Z   to the compiler, or to the compiler name if it is in the PATH.
2021-02-08T22:07:46.6635574Z 
2021-02-08T22:07:46.6635749Z 
2021-02-08T22:07:46.6653618Z CMake Error at model-drivers/DUNN__MD_292677547454_000/CMakeLists.txt:40 (project):
2021-02-08T22:07:46.6654247Z   The CMAKE_C_COMPILER:
2021-02-08T22:07:46.6654509Z 
2021-02-08T22:07:46.6655263Z     /home/conda/feedstock_root/build_artifacts/kim-api_1607958627046/_build_env/bin/x86_64-conda-linux-gnu-cc
2021-02-08T22:07:46.6655664Z 
2021-02-08T22:07:46.6656040Z   is not a full path to an existing compiler tool.
2021-02-08T22:07:46.6656297Z 
2021-02-08T22:07:46.6656719Z   Tell CMake where to find the compiler by setting either the environment
2021-02-08T22:07:46.6657275Z   variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
2021-02-08T22:07:46.6657788Z   the compiler, or to the compiler name if it is in the PATH.
2021-02-08T22:07:46.6658065Z 
2021-02-08T22:07:46.6658411Z 
2021-02-08T22:07:46.6674387Z CMake Error at model-drivers/DUNN__MD_292677547454_000/CMakeLists.txt:40 (project):
2021-02-08T22:07:46.6675044Z   The CMAKE_Fortran_COMPILER:
2021-02-08T22:07:46.6675295Z 
2021-02-08T22:07:46.6676081Z     /home/conda/feedstock_root/build_artifacts/kim-api_1607958627046/_build_env/bin/x86_64-conda-linux-gnu-gfortran
2021-02-08T22:07:46.6676467Z 
2021-02-08T22:07:46.6676854Z   is not a full path to an existing compiler tool.
2021-02-08T22:07:46.6677116Z 
2021-02-08T22:07:46.6677535Z   Tell CMake where to find the compiler by setting either the environment
2021-02-08T22:07:46.6678068Z   variable "FC" or the CMake cache entry CMAKE_Fortran_COMPILER to the full
2021-02-08T22:07:46.6678608Z   path to the compiler, or to the compiler name if it is in the PATH.
2021-02-08T22:07:46.6678896Z 
2021-02-08T22:07:46.6679066Z 
2021-02-08T22:07:46.6702160Z -- Configuring incomplete, errors occurred!
```

**Comment by jan-janssen** [comment]
@ellio167 It took me quite some time to understand this one, but the issue is that it looks for the compilers in the path used during the compilation of `kim-api` in this case `/home/conda/feedstock_root/build_artifacts/kim-api_1607958627046` but the compilers are located in `/home/conda/feedstock_root/build_artifacts/openkim-models_1612822607478/_build_env`.

**Comment by ellio167** [comment]
I see.  Yes the setup tries to use the same compilers as were used to build the kim-api.  It can be over written with some special cmake variables.  I can take a look at it a little later....

Ryan

Sent from my iPhone6s
https://z.umn.edu/relliott
OpenPGP: https://z.umn.edu/relliott_asc (576D4997C4D51D92)

**Comment by jan-janssen** [comment]
@ellio167 It seems to work with the new patch, let's see if the tests pass.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,9 +22,9 @@ requirements:
     - make
     - vim
   host:
-    - kim-api
+    - kim-api >=2.2.1
   run:
-    - kim-api
+    - kim-api >=2.2.1
```
Comment (jan-janssen) [inline recipe/meta.yaml:27]:
```suggestion
    - {{ pin_run_as_build(kim-api, min_pin='x.x', max_pin='x.x') }}
```

Comment (jan-janssen) [reply recipe/meta.yaml:27]: @ellio167 if the Openkim-models are compiled with one version of the kim-api is the version of the kim-api allowed to change or should it always be exactly the same?

**Comment by jan-janssen** [comment]
I merge this pull request now - we can still discuss the pinning in a separate pull request.

**Pull request merged**